src/sql.rs

```diff
@@ -1047,10 +1047,8 @@ pub fn housekeeping(context: &Context) {
                     unreferenced_count,
                     entry.file_name()
                 );
-                unsafe {
-                    let path = entry.path().to_c_string().unwrap();
-                    dc_delete_file(context, path.as_ptr());
-                }
+                let path = entry.path().to_c_string().unwrap();
```

Comment:
I realise this isn't your doing per se, but you're messing with this code so... `.unwrap()` here is wrong since housekeeping shouldn't fail like this.  If the dc_delete_file() signature is changed to take Path then none of these conversion need to happen and this would stay `Path`.

Reply:
I am just removing `unsafe` here without changing the code, so it is out of scope of this PR too.

There are many functions in `dc_tools` that better take `AsRef<Path>` instead of strings, that could be next step in `dc_tools` cleanup.